### ISSUE TITLE
conditional broadcast missing from locks module

### ISSUE THREAD

**Issue by planetis-m** [nim-lang/Nim#17669]
### Summary

`broadcast` is used to wake up blocked threads that are placed in a queue. `signal` on the other hand unblocks a single thread. Some applications require that all threads are [awaken](
http://www.qnx.com/developers/docs/qnxcar2/topic/com.qnx.doc.neutrino.getting_started/topic/s1_procs_Condvar_signal_vs_broadcast.html) . This api is currently missing.

**Comment by ringabout** [comment]
Yeah, `SysCond` use `Event` in windows which makes it unnatural to implement `broadcast`.

I ever reimplemented it with `RTL_CONDITION_VARIABLE`.
https://gist.github.com/xflywind/bf90bb478b4f5c7f062717d64cf557bd

**Comment by ringabout** [comment]
Duplicate of https://github.com/nim-lang/Nim/issues/7998

**Comment by planetis-m** [comment]
That's awesome, thanks for looking into it. Any plans on upstreaming it?

**Comment by ringabout** [comment]
yeah, will make a PR soon.
